src/process/mod.rs

```diff
@@ -1538,8 +1542,9 @@ impl std::iter::Iterator for ProcessesIter {
             match self.inner.next() {
                 Some(Ok(entry)) => {
                     if let Ok(pid) = i32::from_str(&entry.file_name().to_string_lossy()) {
-                        if let Ok(proc) = Process::new(pid) {
-                            break Some(Ok(proc));
+                        match Process::new_with_root(self.root.join(pid.to_string())) {
+                            Ok(proc) => break Some(Ok(proc)),
+                            Err(e) => break Some(Err(e)),
```

Comment:
The documented behavior of the `all_processes` function is that `Process` objects that fail to be constructed won't be returned in the iterator.  Your code here would modify that, and I don't think I want to do that right now.

If you have a use-case where you want to know about these failures, please let me know.

Reply:
It was useful only to find why I didn't have all the processes I expected (in my case `/proc` has less process than `/proc-full`)
With this fix it is not needed anymore, I remove it.